docs/LabHome/Onboarding/coding-guide.md

```diff
@@ -0,0 +1,91 @@
+
+
+### Data wrangling
+The data analysis part after (pre)processing often involve data wrangling. We **highly** recommend using R to handle the data wrangling part. [Tidy data](https://r4ds.had.co.nz/tidy-data.html) is often very helpful here. It is important to note that, although some columns seem redundant in the tidy data, it is very important to include them so that future data wrangling will be easy, and it'll be easy for sanity checks too (i.e., you'll know the exact information of the observation, e.g., from which subject, which session, which method, etc)
+
+### Reduce manual work and additional brain work!!!
+This is very important. Manual work e.g., manually counting or changing numbers increases risk of mistakes and reduces the confidence of what you reported in the manuscript. In addition, it will also increase the burden for the replication buddy - they need to more carefully check it to make sure what you reported are consistent with the original output, especially if they are in different format.
+
+* Generate the final table or figure **directly** from the code as much as possible
+  * Avoid additional manual work (e.g., round up, copy-paste, combination of figure panels) to generate the final table or figure you'll use in the paper. Instead:
+  * Consider using R to export pdf of tables.
+  * Consider writing code to directly generate the figure you want, instead of manually adjusting the fonts in Illustrator or Powerpoint later.
+  * Consider writing code to directly combine figure panels and save the figure (rather than use Illustrator to do so).
+  * Consider directly generating the final format of the values, e.g., let the code round up statistics; if you will report R-squared, let the code calculate the squared number for you; let the code count for you, instead of doing it by yourself
+
+## After you coded a bit
+
+### Test out!
+
+#### Use toy data to test out
+It's important to test out if your code really works as you thought. Make some simple toy data, apply your code, and see if you can get what you want.
+
+#### Include sanity checks
+Are the shape (number of rows and columns) of the data frame you generated is consistent with expectations? Use `testthat` from R (e.g., `expect_that()`), and `assert` from Python to assert that what you actually get is the same as what you expect. Sanity checks are very helpful to rule out mistakes.
+
+#### "Make incremental changes and test as you go."
+This section is copied from [an OHBM blog](https://www.ohbmtrainees.com/blog-overview/2023/2/10/coding-best-practices-for-academia-bridging-the-gap-between-research-and-industry):
+
+"Instead of writing your entire code at once, make small changes and test each step as you go. This approach will make debugging much easier. This is especially important if you are using version control software, such as Git, where each incremental change should be its own commit."
```

Comment:
```suggestion
> "Instead of writing your entire code at once, make small changes and test each step as you go. This approach will make debugging much easier. This is especially important if you are using version control software, such as Git, where each incremental change should be its own commit."
```